### DISCUSSION TITLE
Infinite query --> How do i force a refetch

### DISCUSSION THREAD

**Discussion by nikitph** [TanStack/query#4575] [General]
Firstly thanks for the wonderful library. I am having a hard time getting the query to fetch on demand. Here is my query
` const { isLoading, data, fetchNextPage } = useInfiniteQuery('feed', nonProfitApi.fetchFeed, {
    getNextPageParam: () => {
      return pageNo;
    },
    refetchInterval: 10000,
  });`

and here is how i m try to force a refetch (unsuccessfully)
` useEffect(() => {
    return () => {
      if (isRefreshing) {
        qc.invalidateQueries('feed').then(res => setPageNo(1));
      }
    };
  }, [isRefreshing, qc]);`

**Comment by nikitph** [comment]
any ideas would be welcome

**Reply by TkDodo** [reply]
please show a codesandbox reproduction

**Reply by nikitph** [reply]
this is filed under discussion because i know there is a method to do it but i cannot find it. I m not reporting a bug to require a reproduction unless the code i have pasted is exactly how it should be implemented and its a bug in which case i will file it under bugs.

**Reply by TkDodo** [reply]
it's a discussion because it's likely not a bug in react-query itself, but something in the usage is not right. Anyways, it is surely not required to provide a reproduction, but I cannot help without one. Maybe someone else wants to take a look at it :)

**Reply by nikitph** [reply]
Cool. Thanks for the help so far. For anyone else reading this here is my question. where is this use case documented ? The problem statement put simply is 'How to force a full refetch for an infinite query'? I am not looking for a code fully solving the problem. I am fully capable of doing it myself. Merely looking for a pointer in the right direction.

**Reply by ktriviso** [reply]
Hey @nikitph were you ever able to solve this? I have a similar problem where I am trying to invalidateQueries or even refetchQueries but I've been unsuccessful. I have 1 component that has the useInfiniteQuery and another component that is calling the cache to be cleared (does not have access to simply call refetch on the useInfiniteQuery). I've posted in discussions, discord, etc and haven't gotten any responses so any help is much appreciated.

**Reply by nikitph** [reply]
@ktriviso  it's been a while so i dont remember the exact solution but it was probably a combination of setQueryData(null) plus refetch. Is this something you can try? Lmk if this doesn't help and i will try to dig up the code

**Reply by nikitph** [reply]
@ktriviso  it's been a while so i dont remember the exact solution but it was probably a combination of setQueryData(null) plus refetch. Is this something you can try? Lmk if this doesn't help and i will try to dig up the code

**Reply by ktriviso** [reply]
@nikitph I'll give that a try, thanks for getting back to me!
